_episodes/28.md

```diff
@@ -0,0 +1,257 @@
+
+Manfred's additional notes:
+
+* Misc performance and memory usage improvements
+* `SHOW ROLES` fix
+* `EXPLAIN ANALYZE` fix for estimate display
+*  Numerous improvements for Parquet files in Hive and Iceberg connectors
+
+ 
+More info at <https://trino.io/docs/current/release/release-363.html>.
+
+## Concept of the week: Event Stream abstractions and Pravega
+
+### Events and Streams
+
+What is an event? This sounds like a silly question when asked generally. The
+answer is less clear when discussing event-driven systems though. An **event**
+is an action or occurrence that is captured by either a sensor, or a generated
+by a source system, and emitted to a sink system. Some examples include user
+events from an application, system events in telemetry systems, or sensor events
+from monitoring applications.
+
+What is an event stream? Now knowing what an event is, an **event stream** is, an 
+unbounded set of events that are tracked and retained with some notion of time.
+
+The simple view of an event stream will contain a sequential list of events. The
+list contains events that have been processed, and some that still need to be 
+processed.
+
+<p align="center">
+<img align="center" width="75%" height="100%" src="/assets/episode/28/event-stream.png"/><br/>
+Cloud Native Computing Foundation Presentation: <a href="https://www.cncf.io/wp-content/uploads/2020/08/pravega-overview-cncf-apr-2020.pdf">Source</a>.
+</p>
+
+This is very different from a more realistic view of event streams that 
+considers that events arrive and are processed in parallel. Event load may also
+fluctuate as events may burst around specific events or events have specific 
+periodic behavior. While taking event ingest (writes) into consideration, it is
+also important to consider event egress (reads) as part of the problem of 
+representing event streams.
+
+<p align="center">
+<img align="center" width="75%" height="100%" src="/assets/episode/28/event-stream-realistic.png"/><br/>
+Cloud Native Computing Foundation Presentation: <a href="https://www.cncf.io/wp-content/uploads/2020/08/pravega-overview-cncf-apr-2020.pdf">Source</a>.
+</p>
+
+### Pravega and segments
+
+Engineers at Dell labs wanted to find a better abstraction to solve for the 
+problems they saw in existing event streaming systems. This included how to 
+address this type of constant shift in scaling while also addressing the brittle
+storage abstractions that even streams use today. This storage abstraction needs 
+to allow for both realtime and historical analytics. The data along a particular
+transaction also needs to be consistent.
+
+<p align="center">
+<img align="center" width="50%" height="100%" src="/assets/episode/28/segment.png"/><br/>
+Cloud Native Computing Foundation Presentation: <a href="https://www.cncf.io/wp-content/uploads/2020/08/pravega-overview-cncf-apr-2020.pdf">Source</a>.
+</p>
+
+
+Their solution is Pravega and the core of Pravega build around modeling streams
```

Comment:
Pravega. The core of Pravega models streams built around a storage unit. called a segment.